x/data-proxy/keeper/keeper.go

```diff
@@ -79,6 +79,16 @@ func (k Keeper) processProxyFeeUpdate(ctx sdk.Context, pubKeyBytes []byte, proxy
 		UpdateHeight: updateHeight,
 	}
 
+	// Check if the max updates per block is reached
+	keys, err := k.getFeeUpdateKeys(ctx, updateHeight)
+	if err != nil {
+		return 0, err
+	}
+
+	if len(keys) >= types.MaxUpdatesPerBlock {
+		return 0, types.ErrMaxUpdatesReached
+	}
+
 	// Delete previous pending update, if applicable
 	if proxyConfig.FeeUpdate != nil {
```

Comment:
Should this case be considered separately at the earlier stage if our intention is to always allow updating an existing  fee update?

Reply:
I think we want to only allow rescheduling if the new update height is not already full, otherwise an attacker could still fill one height with a lot of updates. Or am I misunderstanding your comment?

Reply:
I just thought rescheduling could be allowed because it wouldn't create a new entry.

Reply:
I think I'm missing something :D 

How I understand the scenario:
1. For block 100 we have 25 (max) updates scheduled.
2. Proxy A schedules an update for block 99, which now has 1 update scheduled.
3. Proxy A reschedules from 99 to 100, which would delete the entry at 99 and create one at 100 leaving us with 26 scheduled for block 100.

So in reality the order won't matter since the deletion of the update on 99 should revert once we error upon 'discovering' that there are already 25 updates scheduled for 100 leaving us with the same outcome. I just find the flow a little more natural to first check if we can 'reschedule' before deleting the previous entry.

Reply:
Ahh thanks for explaining. I was assuming rescheduling at the same block height as before.
In an unlikely scenario when the reschedule happens at the same height as before, I guess this will prevent an update even though the max won't be exceeded? I guess we can ignore this scenario, though.

Reply:
Aaah yes, in that scenario it'll fail but if I'm putting on my contractor cap "it's not a bug, it's a feature! we're saving the user Txs fees" 😂